config/test.exs

```diff
@@ -1,8 +1,7 @@
 import Config
 
-config :redis_mutex,
-  lock_module: RedisMutex.LockMock,
-  redis_options: [
-    host: "localhost",
-    port: 6379
-  ]
+config :redis_mutex, RedisMutex, redis_url: "redis://localhost:6379"
```

Comment:
do we still need those in test since we are moving away from that config?

Reply:
I've updated how the configuration is used, but I've still left an option to set things in the config. My thinking here is that it is a useful alternative to setting these values directly when adding `RedisMutex` to the supervision tree.

Reply:
no that s actually part of the anti pattern. Configuration like those should be on the application of whoever needs it at the config. The onyl thing that matter is to actually pass the options to the supervision tree option part. And have a good sane default that does not need  options to work

Reply:
What would that look like in this case? Would something like this be passed in when adding the app to the suprvision tree?

```elixir
{RedisMutex, otp_app: :my_app}
```

And then in `RedisMutex`'s start_link, something like this?

```elixir
 def start_link(start_options) do
  otp_app = Keyword.get(start_options, :otp_app)
  config_opts = Application.get_env(otp_app, RedisMutex)
  ...
  end
```

Reply:
@epinault 

I have updated this so that none of the options are set in the config. They must be provided when putting `RedisMutex` in the supervision tree.